typescript-local-storage/data.ts

```diff
@@ -0,0 +1,29 @@
+/* exported todos */
+interface Todo {
+  todoId: string;
+  task: string;
+  isCompleted: boolean;
+}
+
+let todos: Todo[] = [];
+
+function writeTodos(): Todo {
+  const todosJSON = JSON.stringify(todos);
+  localStorage.setItem('todos-storage', todosJSON);
+  console.log('the type of', typeof todosJSON);
+  console.log('todosJSON', todosJSON);
+}
+writeTodos();
+
+function readTodos(): Todo {
+  const readTod = localStorage.getItem('todos-storage');
+  console.log('readTod', readTod);
+  console.log('the type of', typeof readTod);
+  if (readTod === 'string') {
+    const todoIDJSON = JSON.parse(readTod);
+    console.log('todoIDJSON', todoIDJSON);
+  } else;
+  const todos = readTodos();
+  console.log('todos', todos);
```

Comment:
You don't want to check if `readTod` is equal to a `'string'`, you just want to check if it has a value. Here is how your function should be written instead
```suggestion
function readTodos(): Todo[] {
  const todosJSON = localStorage.getItem('todos-storage');
  if (todosJSON) {
    return JSON.parse(todosJSON);
  }
  return [];
}
```